dev_guide/templates.adoc

```diff
@@ -671,12 +676,14 @@ The following is an example of different objects' fields being exposed:
 ----
 kind: Template
 apiVersion: v1
+metadata:
+  name: my-template
 objects:
 - kind: ConfigMap
   apiVersion: v1
   metadata:
```

Comment:
Need define objects.metadata.name here

Reply:
@xiuwang PTAL

Reply:
@mburke5678 Other objects in this template also need add objects.metadata.name

Reply:
@xiuwang I added an objects.metadata.name` to each object in the template. Is name required? The `oc create -f`appears to have worked without the names.

Reply:
@mburke5678 No error when 'oc create -f ' without objects.metadata.name, but will prompt error when 'oc new-app <above_template>'
 error: ConfigMap "" is invalid: metadata.name: Required value: name or generateName is required
    error: Secret "" is invalid: metadata.name: Required value: name or generateName is required
    error: Service "" is invalid: metadata.name: Required value: name or generateName is required
    error: Route "" is invalid: [metadata.name: Required value: name or generateName is required, spec.path: Invalid value: "mypath": path must begin with /, spec.to.name: Required value]

Reply:
@bparees do you view the above as an incompatibility in new-app which we should look at fixing?

Reply:
@jim-minter @bparees @xiuwang If it matters, I was running the command on a 3.5 system. I can try on 3.7 once I get it installed.

Reply:
@xiuwang I did update the PR with object names for secret, service, and route.

Reply:
@jim-minter i think we should first determine which version of oc client it effects.  "oc create -f" is effectively doing server side validation.  "oc new-app" is effectively doing client side validation, so if you're using a 3.5 client and a 3.7 server, you're getting different validation versions.

Reply:
More coffee needed.  I don't think there's a bug here.  `oc create <template object>` doesn't verify that objects[*].metadata.name exist (in any case, any could be parameterised, and the parameter could be left empty at process time).  `oc process <template object> | oc create -f -` will error out if any object has no name.  `oc new-app` is correctly effectively doing the latter.

Reply:
+1

Reply:
(I missed the nuance of what object was actually being created also)

Reply:
@mburke5678 Thanks all of  you, LGTM now